warp/Network/Wai/Handler/Warp.hs

```diff
@@ -371,3 +391,10 @@ setHTTP2Disabled y = y { settingsHTTP2Enabled = False }
 -- Since 3.0.10
 pauseTimeout :: Request -> IO ()
 pauseTimeout = fromMaybe (return ()) . Vault.lookup pauseTimeoutKey . vault
+
+-- | Getting file information from the inside cache.
+--   If file information is not available, an IO error is thrown.
+--
+-- Since 3.1.10
+getFileInfo :: Request -> FilePath -> IO FileInfo
+getFileInfo = fromMaybe (\_ -> throwIO (userError "getFileInfo")) . Vault.lookup getFileInfoKey . vault
```

Comment:
Thank you for your advice. I pushed a new comment. I think that this pull request is ready to merge.